BlueprintUI/Sources/Blueprint View/BlueprintView.swift

```diff
@@ -28,33 +28,52 @@ import UIKit
 public final class BlueprintView: UIView {
 
     private var needsViewHierarchyUpdate: Bool = true
-    private var hasUpdatedViewHierarchy: Bool = false
+    private var nextViewHierarchyUpdateEnablesAppearanceTransitions: Bool = false
+    
     private var lastViewHierarchyUpdateBounds: CGRect = .zero
 
     /// Used to detect reentrant updates
     private var isInsideUpdate: Bool = false
 
     private let rootController: NativeViewController
-
+    
     /// The root element that is displayed within the view.
     public var element: Element? {
-        didSet {
-            setNeedsViewHierarchyUpdate()
+        set {
+            self.set(element: newValue)
         }
+        get {
+            return self._element
+        }
+    }
+    
+    private var _element : Element?
+    
+    public func set(animated : Bool = BlueprintView.shouldAnimateChanges, element: Element?)
+    {
+        _element = element
+        
+        self.nextViewHierarchyUpdateEnablesAppearanceTransitions = animated
```

Comment:
Consider merging this with `needsViewHierarchyUpdate` via an enum instead of two bools